plugins/modules/aci_fabric_external_connection_profile.py

```diff
@@ -0,0 +1,302 @@
+  type: str
+  sample: ?rsp-prop-include=config-only
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec, aci_annotation_spec, aci_owner_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(aci_owner_spec())
+    argument_spec.update(
+        description=dict(type="str", aliases=["descr"]),
+        fabric_id=dict(type="int", aliases=["id", "fabric"]),
+        name=dict(type="str", aliases=["profile_name"]),
+        community=dict(type="str", aliases=["rt", "route_target"]),
+        site_id=dict(type="int", aliases=["sid", "site", "s_id"]),
+        peering_type=dict(type="str", aliases=["p_type", "peer", "peer_t"]),
+        peering_password=dict(type="str", aliases=["peer_password", "peer_pwd"], no_log=True),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["fabric_id"]],
+            ["state", "present", ["fabric_id"]],
+        ],
+    )
+
+    aci = ACIModule(module)
+
+    description = module.params.get("description")
+    fabric_id = module.params.get("fabric_id")
+    name = module.params.get("name")
+    community = module.params.get("community")
+    peering_type = module.params.get("peering_type")
+    peering_password = module.params.get("peering_password")
+    site_id = module.params.get("site_id")
+    state = module.params.get("state")
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class="fvFabricExtConnP",
+            aci_rn="tn-infra/fabricExtConnP-{0}".format(fabric_id),
+            module_object=fabric_id,
+            target_filter={"id": fabric_id},
+        ),
+        child_classes=["fvPeeringP"],
+    )
+
+    aci.get_existing()
+
+    if state == "present":
+        aci.payload(
+            aci_class="fvFabricExtConnP",
+            class_config=dict(
+                descr=description,
+                id=fabric_id,
+                name=name,
+                rt=community,
+                siteId=site_id,
+            ),
+            child_configs=[{"fvPeeringP": {"attributes": {"type": peering_type, "password": peering_password}}}],
```

Comment:
It is not actually possible to delete via APIC UI. So I don't think we should be able to delete this child. 
The peering type needs to be set to one or the other options.
Although the UI is a bit buggy, I don't belive this child is designed to be deleted from the profile.
I did have to change the option to include a default for peering_type because not setting the type doesn't default in the APIC which appears to cause invalid config.